src/commands/actions/KeySharesAction.ts

```diff
@@ -64,35 +69,66 @@ export class KeySharesAction extends BaseAction {
   private async processKeystorePath(): Promise<KeySharesItem[]> {
     const keystorePath = sanitizePath(String(this.args.keystore).trim());
     const { files } = await getKeyStoreFiles(keystorePath);
-    const validatedFiles = await this.validateKeystoreFiles(files);
 
-    const singleKeySharesList = await Promise.all(validatedFiles.map((file, index) =>
-      this.processFile(file, this.args.password, this.getOperators(), this.args.owner_address, this.args.owner_nonce + index)
+    const keystoreSet : KeystoresArgs[] = [];
+
+    for (const file of files) {
+      const isDir = (await fsp.stat(file)).isDirectory();
+      let keystoreFile = file;
+      let keystorePassword = '';
+
+      if (isDir) {
+        const dir = await fsp.opendir(file);
+        for await (const dirent of dir) {
+          if (dirent.name.includes('keystore')) {
+            keystoreFile = path.join(file, dirent.name);
+          }
+          if (dirent.name.includes(this.args.password)) {
+            keystorePassword = await fsp.readFile(path.join(file, dirent.name), 'utf-8');
+          }
+        }
+      }
+      else {
+        keystorePassword = await fsp.readFile(this.args.password, 'utf-8');
```

Comment:
move this line outside of for loop in line 75

Reply:
@IlyaVi If this line is moved outside the for loop at line 75, there will be an issue with setting the value of keystoreSet.